presto-main/src/main/java/com/facebook/presto/operator/HashGenerator.java

```diff
@@ -15,22 +15,13 @@
 
 import com.facebook.presto.spi.Page;
 
-import static com.google.common.base.Preconditions.checkState;
-
 public interface HashGenerator
 {
     long hashPosition(int position, Page page);
 
     default int getPartition(int partitionCount, int position, Page page)
     {
         long rawHash = hashPosition(position, page);
-
-        // clear the sign bit
-        rawHash &= 0x7fff_ffff_ffff_ffffL;
-
-        int partition = (int) (rawHash % partitionCount);
-
-        checkState(partition >= 0 && partition < partitionCount);
-        return partition;
+        return (int) ((Integer.toUnsignedLong(Long.hashCode(rawHash)) * partitionCount) >> 32);
```

Comment:
This is the same trick as https://github.com/prestodb/presto/pull/11832 right? Is this computing `floorMod`? 

Also add some comment stating what this line tries to do? similar to https://github.com/prestodb/presto/blob/de1dda7a12921d65fdeb320ef266bd20f4efb24a/presto-spi/src/main/java/com/facebook/presto/spi/block/MapBlockBuilder.java#L553-L558

Reply:
Sure. Added it as

```
        // This function reduces the 64 bit rawHash to [0, hashTableSize) uniformly. It first reduces the rawHash to 32 bit 
        // integer x then normalize it to x / 2^32 * partitionCount to reduce the range of x from [0, 2^32) to [0, partitionCount) 
```

Reply:
Sure. Added the following
```

        // This function reduces the 64 bit rawHash to [0, hashTableSize) uniformly. It first reduces the rawHash to 32 bit
        // integer x then normalize it to x / 2^32 * partitionCount to reduce the range of x from [0, 2^32) to [0, partitionCount)
        return (int) ((Integer.toUnsignedLong(Long.hashCode(rawHash)) * partitionCount) >> 32);
```